ateam_kenobi/src/spatial/layers/in_field.hpp

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 A Team
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#ifndef SPATIAL__LAYERS__IN_FIELD_HPP_
+#define SPATIAL__LAYERS__IN_FIELD_HPP_
+
+#include "spatial/spatial_layer_factory.hpp"
+
+namespace ateam_kenobi::spatial::layers
+{
+
+class InField : public SpatialLayerFactory {
+public:
+  InField()
+  : SpatialLayerFactory("InField") {}
+
+  void FillLayer(cv::Mat & layer, const World & world) override
+  {
+    SetupLayer(layer, CV_8UC1);
+    if(layer.size() == prev_size) {
```

Comment:
If for some reason the field boundary width changed but the total field size didn't this wouldn't trigger the layer to update.